src/main/java/backend/IssueMetadata.java

```diff
@@ -33,6 +33,15 @@ public IssueMetadata(IssueMetadata other) {
         this.isUpdated = other.isUpdated;
     }
 
+    // Copy constructor used in reconciliation
+    public IssueMetadata(IssueMetadata other, boolean isUpdated) {
+        this.events = new ArrayList<>(other.events);
+        this.comments = new ArrayList<>(other.comments);
+        this.nonSelfUpdatedAt = other.nonSelfUpdatedAt;
+        this.nonSelfCommentCount  = other.nonSelfCommentCount;
+        this.isUpdated = isUpdated;
```

Comment:
Yup. I think more comments wouldn't hurt, though, so I'll clarify its use in a comment.